### PR TITLE
Remove unused NPM modules

### PR DESCRIPTION
**Pull request by ehelms**
This is a large PR as there are a large number of nodejs modules. This updates just the Foreman modules for now to a spec that can support building for the SCL. These changes come with some updated bundle tarballs, and updated dependencies for those bundles. This also removes two packages that were not required by any existing spec: nodejs-select2 and nodejs-jquery-flot. I didn't feel like making 150+ PRs for each package. So my hope is if this succeeds we feel confident enough to merge and fix up any issues after.

### CODE REVIEW

**Comment by ehelms** [comment]
Also, these updates are based on https://github.com/dLobatog/npm2rpm/pull/58

**Comment by ehelms** [comment]
This now only represents about half the modules in an attempt to get it through CI.

**Comment by ehelms** [comment]
[test rpm]

**Comment by ehelms** [comment]
[test rpm]

**Comment by ehelms** [comment]
This PR is now focused on removing unused NPM modules. This took a tactical approach to identify unused modules by analyzing Requires within all spec files. The scripts I used:

```
#!/usr/bin/env ruby
  
BASE_DIR = "packages/foreman/"

dirs = Dir.glob("#{BASE_DIR}nodejs-*/")

dirs = dirs.collect do |package|
  `basename #{package}`
end

dirs = dirs.select do |package|
  npm_module = "npm\\(#{package.gsub('nodejs-', '').rstrip()}\\)"
  found = `grep -r #{npm_module} packages/`
  found = found.split("\n")
  found = found.select { |found_pkg| !found_pkg.include?('bundled') }

  npm_module = "npm\\(@#{package.gsub('nodejs-', '').rstrip().split("-").first}"
  found_grouped = `grep -r #{npm_module} packages/`
  found_grouped = found_grouped.split("\n")

  found = found + found_grouped

  if found.empty?
    true
  else
    false
  end
end

dirs.sort.each do |package|
  puts package
  `git rm -rf #{BASE_DIR}/#{package}`
end
```

File: package_manifest.yaml [hunk 1]
```diff
@@ -223,153 +222,131 @@ foreman_nodejs_packages:
...
-    nodejs-babel-plugin-syntax-dynamic-import: {}
+    nodejs-babel-cli:
+      strategy: bundle
+    nodejs-babel-core:
+      strategy: bundle
+    nodejs-babel-loader:
+      strategy: bundle
+    nodejs-babel-plugin-lodash:
+      strategy: bundle
+    #nodejs-babel-plugin-syntax-dynamic-import: {}
```
Comment (ekohl) [inline package_manifest.yaml]: Nit: should this be removed?

**Comment by ehelms** [comment]
The work I started in Obal to add an `update-spec` command (https://github.com/theforeman/obal/pull/207) uses the strategy and name as input.

**Pull request merged**